lib/AST/ASTDumper.cpp

```diff
@@ -2926,10 +2926,16 @@ static void dumpProtocolConformanceRec(
       }
     }
 
-    for (auto requirement : normal->getConditionalRequirements()) {
+    if (auto condReqs = normal->getConditionalRequirementsIfAvailable()) {
```

Comment:
Hmm. It's unfortunate that the dumper can modify state here by triggering the computation of conditional requirements.

Reply:
Good point.

Reply:
Stopped the mutation in these cases.